core/lib/basic_types/src/tee_types.rs

```diff
@@ -1,9 +1,71 @@
+use std::{fmt, str::FromStr};
+
 use serde::{Deserialize, Serialize};
-use strum::{Display, EnumString};
 
-#[derive(Debug, Clone, Copy, PartialEq, EnumString, Display, Serialize, Deserialize)]
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
 #[non_exhaustive]
 pub enum TeeType {
-    #[strum(serialize = "sgx")]
     Sgx,
 }
+
+impl fmt::Display for TeeType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            TeeType::Sgx => write!(f, "sgx"),
+        }
+    }
+}
+
+impl FromStr for TeeType {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        serde_json::from_str(&format!("\"{}\"", s.to_lowercase()))
+            .map_err(|_| format!("Invalid value for TeeType: {}", s))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use serde_json;
+
+    use super::*;
+
+    #[derive(Deserialize)]
+    struct Config {
+        tee_type: TeeType,
+    }
+
+    #[test]
+    fn test_deserialize_teetype() {
+        let json_str = "\"sgx\"";
+        let tee_type: TeeType = serde_json::from_str(json_str).unwrap();
+        assert_eq!(tee_type, TeeType::Sgx);
+
+        for json_str in &["\"Sgx\"", "\"SGX\""] {
+            let result: Result<TeeType, _> = serde_json::from_str(json_str);
+            assert!(result.is_err());
+        }
+    }
+
+    #[test]
+    fn test_enumstring_teetype() {
+        assert_eq!(TeeType::from_str("sgx").unwrap(), TeeType::Sgx);
+        assert_eq!(TeeType::from_str("Sgx").unwrap(), TeeType::Sgx);
+        assert_eq!(TeeType::from_str("SGX").unwrap(), TeeType::Sgx);
+    }
+
+    #[test]
+    fn test_display_teetype() {
+        assert_eq!(TeeType::Sgx.to_string(), "sgx");
+    }
+
+    #[test]
+    fn test_envy_teetype() {
+        std::env::set_var("TEE_TYPE", "sgx");
+
+        let config = envy::from_env::<Config>().unwrap();
+        assert_eq!(config.tee_type, TeeType::Sgx);
+    }
```

Comment:
This test is not needed. It just calls `serde` under the hood.
On a separate note, env variables are global state, so this test may break other tests and can break itself (unless access to env vars is put under a mutes.

Reply:
That test was suggested to be added by @haraldh. I just removed it [here](https://github.com/matter-labs/zksync-era/pull/2798/commits/72f96936edc2a21bf8df579e0d954277fd4dc390). ✅ But to be fair, he also suggested the same code change as you did [here](https://github.com/matter-labs/zksync-era/pull/2798#discussion_r1743537744).